packages/react-native-bpk-component-calendar/src/ios/CalendarBridge/RCTBPKCalendarManager.m

```diff
@@ -105,4 +109,82 @@ - (void)calendar:(RCTBPKCalendar *)calendar didChangeDateSelection:(NSArray<BPKS
     calendar.onDateSelection(@{@"selectedDates": dateArray});
 }
 
+- (RCTBPKColorBucket *_Nullable) colorBuckedForDate:(NSDate *)date colorBuckets:(NSArray<RCTBPKColorBucket *> *)colorBuckets {
+    for (RCTBPKColorBucket *bucket in colorBuckets){
+        switch (bucket.days.matcherType) {
+            case RCTBPKDateMatcherTypeRange:
+                if ([RCTBPKCalendarDateUtils date:date isAfterDate:bucket.days.dates[0]] &&
+                    [RCTBPKCalendarDateUtils date:date isBeforeDate:bucket.days.dates[1]]) {
+                    return bucket;
+                }
+                break;
+            case RCTBPKDateMatcherTypeBefore:
+                if ([RCTBPKCalendarDateUtils date:date isBeforeDate:bucket.days.dates[0]]) {
+                    return bucket;
+                }
+                break;
+            case RCTBPKDateMatcherTypeAfter:
+                if ([RCTBPKCalendarDateUtils date:date isAfterDate:bucket.days.dates[0]]) {
+                    return bucket;
+                }
+                return nil;
+            case RCTBPKDateMatcherTypeAny:
+                for (NSDate *bucketDate in bucket.days.dates) {
+                    if ([bucketDate isEqualToDate:date]) {
+                        return bucket;
+                    }
+                }
+                break;
+        }
+    }
+    return nil;
+}
+
+- (UIColor *)calendar:(BPKCalendar *)calendar fillColorForDate:(NSDate *)date {
+    RCTBPKCalendar *rctCalendar = (RCTBPKCalendar *)calendar;
+    if(rctCalendar != nil) {
```

Comment:
Why would this ever happen? By definition if we are in a delegate method of a calendar the `calendar` parameter will never be `nil`

Reply:
In fact it's not marked as `_Nullable` so it violates our own API

Reply:
Use an assert to check that the calendar is a `RCTBPKCalendar.class`